instaclustr/resource_cluster.go

```diff
@@ -329,33 +341,31 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 func resourceClusterUpdate(d *schema.ResourceData, meta interface{}) error {
 	d.Partial(true)
 	// currently only cluster resize is supported
-	if !d.HasChange("node_size") {
```

Comment:
yes, because I feel it does not make sense. No other provider does this and looking at the terraform howto's they don't fail and only test for changes based on fields:

https://learn.hashicorp.com/tutorials/terraform/provider-update#implement-update

When writing these changes I kept running into this error because it was detecting this new field as a change.

Reply:
It is not a conventional implementation (having said that terraform is still new enough that it doesn't have too many conventions anyways). My only concern is that, say the user decided to update the data centre for the cluster, we would not actually update the data centre, and `terraform plan` would always show there is a change to be made. I think the right way to fix this would be to set ForceNew on all of the arguments except for `node_size`, does that make sense?

Reply:
Yeah, ok. What is the size of the change? Does it make sense to move that into another PR?

Reply:
It would be a matter of adding `ForceNew: true` to each property listed [here](https://github.com/instaclustr/terraform-provider-instaclustr#properties) (child properties excluded). It would be reasonable to move this into another PR but in that case the fail condition you had removed should be brought back for the reasons I've mentioned [above](https://github.com/instaclustr/terraform-provider-instaclustr/pull/51#discussion_r484650801).

Reply:
but for anyone will a current resource, it will always fail. it is why I removed it.

Reply:
```go
	// currently only cluster resize is supported
	if !d.HasChange("node_size") {
		return fmt.Errorf("[Error] The cluster doesn't support update")
	}
```

It would only fail if there is a change to the resource that isn't `node_size`. I just tested it and it only fails when I change one of the properties that isn't `node_size`. My sample configuration - 

```terraform
resource "instaclustr_cluster" "example1" {
  cluster_name = "testcluster"
  node_size = "t3.small"
  data_centre = "US_WEST_2"
  sla_tier = "NON_PRODUCTION"
  cluster_network = "192.168.0.0/18"
  private_network_cluster = false
  cluster_provider = {
    name = "AWS_VPC"
  }
  rack_allocation = {
    number_of_racks = 2
    nodes_per_rack = 1
  }

  bundle {
    bundle = "APACHE_CASSANDRA"
    version = "3.11.4"
    options = {
      auth_n_authz = true
    }
  }
}

```

Reply:
is that using tf13 and this branch?

Reply:
I have used TF12 (since we are currently facing issues with TF13 as mentioned [here](https://github.com/instaclustr/terraform-provider-instaclustr#requirements)) with the master branch to verify that the old logic works.

Reply:
Would like to also add that the existing acceptance test `TestAccCluster` is failing because it expects the update to throw an error - `ExpectError: regexp.MustCompile("The cluster doesn't support update")`

Reply:
Hey there Jürgen, any update/further questions on this?

Reply:
I'll be honest and I don't really agree with you on your approach, but in the end it is your product so you, do you (please feel free to take over this PR). In the short term the custom build we have of the provider works for for us and in the medium term, since AWS MSK now supports [scram](https://aws.amazon.com/about-aws/whats-new/2020/09/amazon-msk-now-supports-sasl-scram-authentication-with-usernames-and-passwords-secured-by-aws-secrets-manager/) we will more than likely move to that.

Thanks

Reply:
Thanks, I'll take it over. As mentioned before the current changes are breaking the acceptance test and causes issues with `terraform plan` so I'll revert just this block.